packages/quantic/force-app/main/default/classes/SampleTokenProvider.cls

```diff
@@ -0,0 +1,9 @@
+global with sharing class SampleTokenProvider implements ITokenProvider {
+    public static String getHeadlessConfiguration() {
```

Comment:
Please add a lot of comments here that explain what that class does, what data it returns, how to change it to connect to your Coveo org instead (like with links to API doc and stuff like that).

This is meant to be sent to devs, let's make sure they have all the information they need to succeed and not ask us questions 😄

Reply:
Exhaustive ToDo:
 - add documentation/example on how to switch to another tokenprovider
 - add documentation on the return type
 - add link to Salesforce documentation for linking
 - In the CustomTokenProvider add a link to create API key.